### PR TITLE
use rapids-build-backend

### PR DESCRIPTION
**Pull request by jameslamb**
Contributes to https://github.com/rapidsai/build-planning/issues/31
Contributes to https://github.com/rapidsai/dependency-file-generator/issues/89

Proposes introducing `rapids-build-backend` as this project's build backend, to reduce the complexity of various CI/build scripts.

### CODE REVIEW

File: ci/build_wheel.sh [hunk 1]
```diff
@@ -8,14 +8,7 @@ source "$(dirname "$0")/install_latest_cuda_toolkit.sh"
...
 
-# This is the version of the suffix with a preceding hyphen. It's used
-# everywhere except in the final wheel name.
-PACKAGE_CUDA_SUFFIX="-${RAPIDS_PY_CUDA_SUFFIX}"
-
-# Patch project metadata files to include the CUDA version suffix.
-sed -i "s/^name = \"pynvjitlink\"/name = \"pynvjitlink${PACKAGE_CUDA_SUFFIX}\"/g" pyproject.toml
-
-rapids-logger "Build wheel"
+rapids-logger "Build wheel"s
```
Comment (bdice) [inline ci/build_wheel.sh]:
```suggestion
rapids-logger "Build wheel"
```

Comment (jameslamb) [inline ci/build_wheel.sh:11]:
```suggestion
rapids-logger "Build wheel"
```

Not sure how this got in there 😬

**Review by bdice — APPROVED** [review]

**Comment by bdice** [comment]
@jameslamb I pushed [7f32e60](https://github.com/rapidsai/pynvjitlink/pull/88/commits/7f32e608a27c2c5c3f9ba106677c1fb63f40856e) to fix some workflow warnings by bumping the shared-workflows from branch-24.04 to branch-24.08. I lumped it into this PR because this is the last PR I want to merge before we make a new release, and I needed to re-run CI anyway due to a random failure. Hope that's alright.

**Comment by jameslamb** [comment]
Yep totally fine! Thanks for the comment explaining that.

**Comment by jameslamb** [comment]
@bdice I manually re-ran that one failing test job, looks like everything's passing now 🎉

**Pull request merged**